### PR TITLE
refactor(install-expo-modules): move over from `expo/expo-cli`

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Moving the actively used `install-expo-modules` over from `expo/expo-cli`.

Counter-part PR: https://github.com/expo/expo-cli/pull/4786

# How

- Copied folder from `expo/expo-cli`
- Updated `jest.config.js`, `tsconfig.json`, `package.json`
- Added changelog
- Bumped packages to latest versions
  - `@expo/config` (`~8.0.0` → `^8.4.0`) 
  - `@expo/config-plugins` (`~6.0.0` → `^7.7.0`) 
  - `@expo/package-manager` (`0.0.56` → `^1.0.3`)
- Updates tests:
  - Replaced `... as jest.Mock<>` → `jest.mocked(...)`
  - Fixed failing test due to `def min_ios_version_supported` → `def self.min_ios_version_supported`

# Test Plan

See if CI passes

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/install-expo-modules/src/plugins/ios/__tests__/fixtures/AppDelegate-updated.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import UIKit
+
+@UIApplicationMain
+class AppDelegate: AppDelegateWrapper, RCTBridgeDelegate {
+  var window: UIWindow?
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    let bridge = reactDelegate.createBridge(delegate: self, launchOptions: launchOptions)!
```
[the remark this answers is not in the document]
Comment (byCedric) [reply packages/install-expo-modules/src/plugins/ios/__tests__/fixtures/AppDelegate-updated.swift:8]: How are you still not silent

[the remark this answers is not in the document]
Comment (byCedric) [reply packages/install-expo-modules/src/plugins/ios/__tests__/fixtures/AppDelegate-updated.swift:8]: I EXCLUDED YOU!

**Comment by byCedric** [comment]
Chatted with @tsapeta, we are ignoring the `swiftlint` errors for now. It's excluded already, but not included in this PR as they are ran from `main`

**Review by tsapeta — APPROVED** [review]

File: packages/install-expo-modules/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+<p align="center">A tool for existing React Native projects to adopt <a href="https://docs.expo.dev/versions/latest/">expo-modules and SDK easier</a>.</p>
+
+<p align="center">
+  <img src="https://flat.badgen.net/packagephobia/install/install-expo-modules">
+
+  <a href="https://www.npmjs.com/package/install-expo-modules">
+    <img src="https://flat.badgen.net/npm/dw/install-expo-modules" target="_blank" />
+  </a>
+</p>
```
Comment (byCedric) [inline packages/install-expo-modules/README.md:14]:
```suggestion
<p align="center">A tool for existing React Native projects to adopt <a href="https://docs.expo.dev/versions/latest/">expo-modules and SDK easier</a>.</p>
```

File: packages/install-expo-modules/src/plugins/ios/__tests__/fixtures/AppDelegate-updated.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (byCedric) [reply packages/install-expo-modules/src/plugins/ios/__tests__/fixtures/AppDelegate-updated.swift:8]: >:C

**Pull request merged**